rpcs3/util/vm_native.cpp

```diff
@@ -125,10 +145,16 @@ namespace utils
 			ptr = static_cast<u8*>(ptr) + (0x10000 - misalign);
 		}
 
-#ifdef MADV_HUGEPAGE
 		if (orig_size % 0x200000 == 0)
-			::madvise(ptr, orig_size, MADV_HUGEPAGE);
-#endif
+		{
+			if constexpr (c_madv_hugepage)
```

Comment:
minor optimization:
check this if constexpr first

Reply:
Honestly, it would make no difference (although I had to rewrite this part anyway). Compiler would throw away an empty check with obvious lack of side effect.